src/plugins/validation/oas3/semantic-validators/security-definitions-ibm.js

```diff
@@ -0,0 +1,192 @@
+          !flows.password &&
+          !flows.clientCredentials
+        ) {
+          errors.push({
+            message:
+              "oauth2 authorization must have required 'flows'. Valid values are 'implicit', 'password', 'clientCredentials' or 'authorizationCode'",
+            path,
+            authId: key
+          });
+        } else if (flows.implicit) {
+          const authorizationUrl = flows.implicit.authorizationUrl;
+          if (!authorizationUrl) {
+            errors.push({
+              message:
+                "oauth2 authorization implicit flow must have required 'authorizationUrl' parameter if type is `implicit` or `authorizationCode`.",
+              path,
+              authId: key
+            });
+          } else if (!flows.implicit.scopes) {
+            errors.push({
+              message:
+                "oauth2 authorization implicit flow must have required 'scopes' property.",
+              path,
+              authId: key
+            });
+          }
+        } else if (flows.authorizationCode) {
+          const authorizationUrl = flows.authorizationUrl;
+          if (!authorizationUrl) {
+            errors.push({
+              message:
+                "oauth2 authorization implicit flow must have required 'authorizationUrl' parameter if type is `implicit` or `authorizationCode`.",
+              authId: key
+            });
+          }
+        } else if (flows && flows.password) {
+          const tokenURL = flows.tokenURL;
+          if (!tokenURL) {
+            errors.push({
+              message:
+                "oauth2 authorization password flow must have required 'tokenUrl' string parameter.",
+              path,
+              authId: key
+            });
+          }
+        } else if (flows === CLIENT_CREDENTIALS) {
+          if (!tokenUrl) {
+            errors.push({
+              message:
+                "oauth2 authorization clientCredentials flow must have required 'tokenUrl' string parameter.",
+              path,
+              authId: key
+            });
+          }
+        }
+      } else if (type === HTTP) {
+        //scheme is required
+        if (!security.scheme) {
+          errors.push({
+            message:
+              'scheme must be defined for type `http` and must be a string',
+            path,
+            authId: key
+          });
+        }
+      } else if (type == OPENID_CONNECT) {
+        const openIdConnectURL = security.openIdConnectUrl;
+        if (
+          !openIdConnectURL ||
+          typeof openIdConnectURL != 'string' ||
+          !validator.isurl(openIdConnectURL)
+        ) {
+          errors.push({
+            message:
+              'openIdConnectUrl must be defined for openIdConnect property and must be a valid URL',
+            path,
+            authId: key
+          });
+        }
+      }
+    }
+    if (
+      flows === AUTHORIZATION_CODE ||
+      flows === PASSWORD ||
+      flows === CLIENT_CREDENTIALS
```

Comment:
I don't think this "flows" check should be at the top level when it only applies to one type of scheme